test/test_linalg.py

```diff
@@ -922,6 +923,90 @@ def test_nuclear_norm_exceptions_old(self, device):
         self.assertRaisesRegex(IndexError, "Dimension out of range", torch.norm, x, "nuc", (0, 2))
 
 
+    @precisionOverride({torch.float: 1e-2, torch.cfloat: 1e-4})
```

Comment:
The absolute tolerance for float32 is surprisingly large. What's going on there?

Reply:
I assume that NumPy could silently convert the input to fp64 and then back to fp32.
CPU tests pass for `1e-3` tolerance. GPU passes only with `1e-2`. It doesn't mean that the decomposition is incorrect, it's all valid, but the individual entries can differ a bit across different libraries/environments.
A similar thing was observed for `inverse` https://github.com/pytorch/pytorch/pull/45034, where the results of cuSOLVER and MAGMA differ a lot for fp32.

Reply:
I think your analysis is probably correct. Would you verify what dtype NumPy computes in and a comment here explaining the absolute tolerance value? 

How significant a loss of precision is 1e-2 here? In terms of relative tolerance, I mean?